provider crashed

panic: invalid argument to Intn

goroutine 231 [running]:
math/rand.(*Rand).Intn(0x1400029a200?, 0x1018c4e00?)
        math/rand/rand.go:180 +0x64
github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/node.SdkCreate(0x1400007e2d0?)
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/Internal/resource/guest/node/sdk.go:38 +0x180
github.com/Telmate/terraform-provider-proxmox/v2/proxmox.resourceVmQemuCreate({0x1018c79a8, 0x14000276af0}, 0x1400029e600, {0x1017542a0?, 0x14000252a00})
        github.com/Telmate/terraform-provider-proxmox/v2/proxmox/resource_vm_qemu.go:757 +0x143c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x14000712c40, {0x1018c7900, 0x14000bf79b0}, 0xd?, {0x1017542a0, 0x14000252a00})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:806 +0xe8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14000712c40, {0x1018c7900, 0x14000bf79b0}, 0x14000dfa4e0, 0x1400029e380, {0x1017542a0, 0x14000252a00})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:937 +0x86c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x14000362000, {0x1018c7900?, 0x14000bf78c0?}, 0x14001384820)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:1153 +0xb08
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x1400027a0a0, {0x1018c7900?, 0x14000bf6fc0?}, 0x1400097e9a0)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:865 +0x2b0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1018941c0?, 0x1400027a0a0}, {0x1018c7900, 0x14000bf6fc0}, 0x1400029e780, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:518 +0x164
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000187200, {0x1018c7900, 0x14000bf6f30}, {0x1018cbb98, 0x14000272480}, 0x14000ca07e0, 0x140005953b0, 0x101e271f8, 0x0)
        google.golang.org/grpc@v1.63.2/server.go:1369 +0xba0
google.golang.org/grpc.(*Server).handleStream(0x14000187200, {0x1018cbb98, 0x14000272480}, 0x14000ca07e0)
        google.golang.org/grpc@v1.63.2/server.go:1780 +0xc80
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.63.2/server.go:1019 +0x8c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 53
        google.golang.org/grpc@v1.63.2/server.go:1030 +0x150

Error: The terraform-provider-proxmox_v3.0.1-rc8 plugin crashed!

--- a/proxmox/Internal/resource/guest/node/schema.go
+++ b/proxmox/Internal/resource/guest/node/schema.go
@@ -43,6 +43,7 @@ func SchemaNodes(guestType string) *schema.Schema {
 		Type:          schema.TypeSet,
 		Optional:      true,
 		Description:   "A list of nodes the " + guestType + " guest may be placed on.",
+		MinItems:      1,
 		ConflictsWith: []string{RootNode},
 		Elem: &schema.Schema{
 			Type: schema.TypeString,

--- a/proxmox/Internal/resource/guest/node/sdk.go
+++ b/proxmox/Internal/resource/guest/node/sdk.go
@@ -1,6 +1,7 @@
 package node
 
 import (
+	"errors"
 	"math/rand"
 	"time"
 
@@ -8,38 +9,46 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-func SdkUpdate(d *schema.ResourceData, current pveAPI.NodeName) pveAPI.NodeName {
+const error_NoNodeConfigured = "no target node specified"
+
+func SdkUpdate(d *schema.ResourceData, current pveAPI.NodeName) (pveAPI.NodeName, error) {
 	if node, ok := d.GetOk(RootNode); ok {
-		return pveAPI.NodeName(node.(string))
+		return pveAPI.NodeName(node.(string)), nil
 	}
 	nodes := d.Get(RootNodes).(*schema.Set).List()
 	currentNode := string(current)
-	if len(nodes) == 1 {
+	switch len(nodes) {
+	case 0:
+		return "", errors.New(error_NoNodeConfigured)
+	case 1:
 		if currentNode != nodes[0].(string) {
-			return pveAPI.NodeName(nodes[0].(string))
+			return pveAPI.NodeName(nodes[0].(string)), nil
 		}
-		return current
+		return current, nil
 	}
 	if inArray(nodes, currentNode) {
-		return current
+		return current, nil
 	}
 	randomIndex := rand.New(rand.NewSource(time.Now().UnixNano())).Intn(len(nodes))
-	return pveAPI.NodeName(nodes[randomIndex].(string))
+	return pveAPI.NodeName(nodes[randomIndex].(string)), nil
 }
 
-func SdkCreate(d *schema.ResourceData) pveAPI.NodeName {
+func SdkCreate(d *schema.ResourceData) (pveAPI.NodeName, error) {
 	if node, ok := d.GetOk(RootNode); ok {
-		return pveAPI.NodeName(node.(string))
+		return pveAPI.NodeName(node.(string)), nil
 	}
 	nodes := d.Get(RootNodes).(*schema.Set).List()
-	if len(nodes) == 1 {
-		return pveAPI.NodeName(nodes[0].(string))
+	switch len(nodes) {
+	case 0:
+		return "", errors.New(error_NoNodeConfigured)
+	case 1:
+		return pveAPI.NodeName(nodes[0].(string)), nil
 	}
 	randomIndex := rand.New(rand.NewSource(time.Now().UnixNano())).Intn(len(nodes))
-	return pveAPI.NodeName(nodes[randomIndex].(string))
+	return pveAPI.NodeName(nodes[randomIndex].(string)), nil
 }
 
-func inArray(nodes []interface{}, current string) bool {
+func inArray(nodes []any, current string) bool {
 	for i := range nodes {
 		if current == nodes[i].(string) {
 			return true

--- a/proxmox/resource_lxc.go
+++ b/proxmox/resource_lxc.go
@@ -511,9 +511,11 @@ func resourceLxcCreate(ctx context.Context, d *schema.ResourceData, meta interfa
 
 	setGuestID := d.Get(vmID.Root).(int)
 
-	targetNode := node.SdkCreate(d)
+	targetNode, err := node.SdkCreate(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
 	var diags diag.Diagnostics
-	var err error
 	var vmr *pveSDK.VmRef
 	if d.Get("clone").(string) != "" { // Clone
 		var sourceVmr *pveSDK.VmRef

--- a/proxmox/resource_vm_qemu.go
+++ b/proxmox/resource_vm_qemu.go
@@ -751,10 +751,13 @@ func resourceVmQemuCreate(ctx context.Context, d *schema.ResourceData, meta inte
 	vmr := duplicateVmr
 
 	var rebootRequired bool
-	var err error
 
 	if vmr == nil { // Create new VM
-		targetNode := node.SdkCreate(d)
+		targetNode, err := node.SdkCreate(d)
+		if err != nil {
+			return append(diags, diag.FromErr(err)...)
+		}
+
 		config.Node = &targetNode
 
 		var guestID *pveSDK.GuestID
@@ -848,7 +851,10 @@ func resourceVmQemuCreate(ctx context.Context, d *schema.ResourceData, meta inte
 	} else { // Forcefully update an existing VM
 		log.Printf("[DEBUG][QemuVmCreate] recycling VM vmId: %d", vmr.VmId())
 
-		targetNode := node.SdkUpdate(d, vmr.Node())
+		targetNode, err := node.SdkUpdate(d, vmr.Node())
+		if err != nil {
+			return append(diags, diag.FromErr(err)...)
+		}
 		client.StopVm(ctx, vmr)
 
 		rebootRequired, err = config.Update(ctx, false, vmr, client)
@@ -915,7 +921,6 @@ func resourceVmQemuUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 
 	config := pveSDK.ConfigQemu{
 		Name:        d.Get("name").(string),
-		Node:        util.Pointer(node.SdkUpdate(d, vmr.Node())),
 		CPU:         cpu.SDK(d),
 		Description: util.Pointer(d.Get("desc").(string)),
 		Pool:        util.Pointer(pveSDK.PoolName(d.Get(pool.Root).(string))),
@@ -942,6 +947,13 @@ func resourceVmQemuUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 		CloudInit:   mapToSDK_CloudInit(d),
 		TPM:         tpm.SDK(d),
 	}
+
+	tmpNode, err := node.SdkUpdate(d, vmr.Node())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	config.Node = &tmpNode
+
 	if len(qemuVgaList) > 0 {
 		config.QemuVga = qemuVgaList[0].(map[string]interface{})
 	}

